apidef/oas/authentication.go

```diff
@@ -167,6 +167,35 @@ func (ss SecuritySchemes) Import(name string, nativeSS *openapi3.SecurityScheme,
 	return nil
 }
 
+func (ss SecuritySchemes) GetBaseIdentityProvider() (res apidef.AuthTypeEnum) {
+	if len(ss) < 2 {
+		return
+	}
+
+	precedence := map[apidef.AuthTypeEnum]int{
+		apidef.AuthToken:     1,
+		apidef.JWTClaim:      2,
+		apidef.OAuthKey:      3,
+		apidef.BasicAuthUser: 4,
+	}
+
+	resBaseIdentityProvider := len(precedence)
+	res = apidef.OAuthKey
+
+	for _, scheme := range ss {
+		if _, ok := scheme.(*Token); ok {
+			return apidef.AuthToken
+		} else if _, ok := scheme.(*JWT); ok {
```

Comment:
Previous code block has function exit, consider dropping else for readability, less nesting